src/sagemaker_training/mpi.py

```diff
@@ -341,6 +341,18 @@ def _create_command(self):
             command.extend(["-x", "FI_PROVIDER=efa"])
             # Use simple protocol to handle the out-of-order data delivery from EFA
             command.extend(["-x", "NCCL_PROTO=simple"])
+            # Further set NCCL_ALGO for modelparallel job to avoid hang, starting from torch2.0.0
+            env = environment.Environment()
```

Comment:
do we have any inherited environment vars from input? I don't think its appropriate to initial the Environment class again

Reply:
Updated it by passing an existing environment variable.